### PR TITLE
Add a couple of fixes to make it possible to natively build Valhalla on Apple Silicon

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
# Issue

Makes it possible to natively build Valhalla on Apple Silicon Macs without Rosetta. Some tests are red though(can be ARM specific), but it can be addressed separately.

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)
 - [ ] If you made changes to the lua files, update the [taginfo](taginfo.json) too.

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: cmake/FindSQLite3.cmake [hunk 1]
```diff
@@ -44,7 +44,7 @@ set(SQLITE3_LIB_SEARCH_PATHS
 if (APPLE)
   # on macOS, we try hard to prefer aftermarket sqlite3 installations, as the system sqlite3
   # disables extension loading, which breaks use of spatialite later on
-  list(INSERT SQLITE3_INCLUDE_SEARCH_PATHS 0 /usr/local/opt/sqlite3/include /usr/local/include)
+  list(INSERT SQLITE3_INCLUDE_SEARCH_PATHS 0 /usr/local/opt/sqlite3/include /usr/local/include /opt/homebrew/opt/sqlite/include)
```
Comment (SiarheiFedartsou) [inline cmake/FindSQLite3.cmake:47]: for some reason `brew` installs packages in different directory on Apple Silicon

Comment (SiarheiFedartsou) [inline cmake/FindSQLite3.cmake:47]: for some reason `brew` installs packages in different directory on Apple Silicon

**Comment by kevinkreiser** [comment]
seems its breaking the x86 osx build? segfault in the utrecht tile building sub commands...

```
[ 93%] Building Utrecht Tiles...
make[2]: *** [test/data/utrecht_tiles/traffic.tar] Segmentation fault: 11
make[1]: *** [test/CMakeFiles/utrecht_tiles.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

since you added the path on the end the list of search paths it makes me suspect that the string you use to detect the x86 variant of mac hardware must not be correct? if i have a chance to night i can ssh in to see

File: CMakeLists.txt [hunk 2]
```diff
@@ -325,8 +325,8 @@ if(ENABLE_DATA_TOOLS)
     set_target_properties(${program} PROPERTIES FOLDER "Data Tools")
     target_include_directories(${program} PUBLIC ${VALHALLA_SOURCE_DIR}/third_party/cxxopts/include)
     target_link_libraries(${program} valhalla)
-    if (LUAJIT_FOUND AND APPLE)
-      # Using LuaJIT on macOS requires a couple of extra linker flags
+    if (LUAJIT_FOUND AND APPLE AND CMAKE_HOST_SYSTEM_PROCESSOR EQUAL "x86_64")
```
Comment (kevinkreiser) [inline CMakeLists.txt:328]: i guess it cant be this!

```
static:~ distiller$ uname -m
x86_64
```

File: cmake/FindSQLite3.cmake (hunk 1 above)
Comment (kevinkreiser) [reply cmake/FindSQLite3.cmake:47]: and this new path doesnt even exist on the CI mac machine so... im not sure why the build is segfaulting!

File: CMakeLists.txt (hunk 2 above)
Comment (SiarheiFedartsou) [reply CMakeLists.txt:328]: Most likely I should use STREQUAL, but not EQUAL here. I'll fix that a bit later.

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**